### ISSUE TITLE
Need any help?

### ISSUE THREAD

**Issue by michaelr524** [sagan-software/rust-eos#1]

**Comment by liamcurry** [comment]
Help would be great! I didn't think anyone would find this repo yet but I'll work on some basic docs tomorrow.

Right now the short term goal is to translate the [`hello` contract](https://github.com/EOSIO/eosio.cdt/blob/c692723166bcc2d21b16f798b06b9c7b4abd8188/examples/hello) and the [`tic_tac_toe` contract](https://github.com/EOSIO/eos/tree/874205c6b0de5b37b750751dfe79c52666beb6b9/contracts/tic_tac_toe) to Rust. That shouldn't be too difficult, but it is a decent goal and will require a lot of low-level setup work that can be used later to build higher level abstractions.

I've got the low-level bindings from [`eosiolib`](https://github.com/EOSIO/eosio.cdt/tree/c692723166bcc2d21b16f798b06b9c7b4abd8188/libraries/eosiolib) being automatically generated in the `eosio_sys` crate. All of the C API functions on the [EOS developers reference page](https://developers.eos.io/eosio-cpp/reference) should be available in that crate.

Now work needs to be done to make these C APIs more usable, which will eventually mean porting most of the C++ code in `eosiolib` to Rust. But for now the C APIs are mostly fine for building a prototype. I think the two biggest missing pieces are:

1. Reading action data (https://github.com/EOSIO/eosio.cdt/blob/c692723166bcc2d21b16f798b06b9c7b4abd8188/libraries/eosiolib/dispatcher.hpp#L65)
2. Reading / writing to tables (https://github.com/EOSIO/eosio.cdt/blob/c692723166bcc2d21b16f798b06b9c7b4abd8188/libraries/eosiolib/multi_index.hpp)

Once we have solutions for those two issues we should be able to build crude working prototypes of the `hello` and `tic_tac_toe` contracts, and then start to revisit/refine the higher-level APIs.

**Comment by liamcurry** [comment]
I added some docs to the README here: https://github.com/sagan-software/rust-eos#contributor-quickstart

**Comment by liamcurry** [comment]
Hey @michaelr524 I moved the contributing docs to `CONTRIBUTING.md`. I also expanded on some of the goals for this project in `README.md` and have a longer example for how to get started writing EOS smart contracts in Rust. Improving docs and making an announcement post is my top priority right now so things should get better on this front soon. In the meantime I'm going to close this issue. Thanks!
